gatsby-source-chec/gatsby/node/sourceNodes.js

```diff
@@ -54,20 +54,34 @@ const sourceNodes = async (
     },
   });
 
-  categories.forEach((category) =>
+  categories.forEach((category) => {
+    const productIds = products.reduce((ids, product) => {
+      const matchingCategory = product.categories.find(
+        (cat) => cat.id === category.id
+      );
+
+      if (!matchingCategory) return ids;
+
+      return [...ids, product.id];
+    }, []);
+
     createNode({
       ...category,
+      products: productIds,
       internal: {
         type: `ChecCategory`,
         content: JSON.stringify(category),
         contentDigest: createContentDigest(category),
       },
-    })
-  );
+    });
+  });
+
+  products.forEach(({ categories, ...product }) => {
+    const categoryIds = categories.map(({ id }) => id);
 
-  products.forEach((product) => {
     createNode({
       ...product,
+      categories: categoryIds,
```

Comment:
So the API returns category IDs that a product belongs to, but this is returning full category objects - is that correct?

Reply:
This is used to link internally with Gatsby nodes for categories. Since the Gatsby nodes use the ID from Chec, we can infer the linking using that. 

We have to do a little bit of workaround to check if the products belong in the category by checking every time inside the array. This is due to the fact categories don’t return products (or anything about their relationships at all) otherwise it would be just the same call for mapping the IDs of products in categories like we have here. 

I hope that makes sense - I can walk you through it if needed.

Reply:
Ah ok, thanks. Makes sense!